src/modules/auth/__tests__/auth.controller.spec.ts

```diff
@@ -0,0 +1,48 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import * as request from 'supertest';
+import { INestApplication } from '@nestjs/common';
+import { AppModule } from '../../app/app.module';
+import { SeedService } from '../../seed/seed.service';
+import { ConfigModule } from '@nestjs/config';
+
+describe('Authentication', () => {
+  let app: INestApplication;
+  let seedService: SeedService;
+
+  beforeAll(async () => {
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        AppModule,
+        ConfigModule.forRoot({
+          envFilePath: `.env.${process.env.NODE_ENV}`,
+        }),
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    await app.init();
+
+    seedService = moduleFixture.get<SeedService>(SeedService);
+
+    // clear and seed the database before running the tests
+    await seedService.clear();
+    await seedService.seed();
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  it('should login with valid credentials', async () => {
+    const staticApiKey = 'FORTESTFORTESTFORTESTFORTESTFORTEST';
+
+    const response = await request(app.getHttpServer())
```

Comment:
This is more of an integration test than it is a unit test.

Let's have a tests folder in the root folder which will house the integration tests like this https://github.com/Greenstand/treetracker-wallet-api/tree/master/__tests__

and the unit tests can be in the auth folder, like this https://github.com/Greenstand/treetracker-wallet-api/blob/master/server/services/AuthService.spec.js